pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -563,7 +563,7 @@ public PulsarClient getReplicationClient(String cluster) {
                                     : data.getServiceUrlTls())
                             .enableTls(true)
                             .tlsTrustCertsFilePath(pulsar.getConfiguration().getBrokerClientTrustCertsFilePath())
-                            .allowTlsInsecureConnection(pulsar.getConfiguration().isTlsAllowInsecureConnection());
+                            .allowTlsInsecureConnection(pulsar.getConfiguration().isBrokerClientTlsAllowInsecureConnection());
```

Comment:
for backward compatibility consideration, shall we implement `isBrokerClientTlsAllowInsecureConnection()`?

basically if brokerClientTlsAllowInsecureConnection is not provided, use tlsAllowInsecureConnection.

Reply:
hmm.. `brokerClientTlsAllowInsecureConnection` is a boolean value and if it's not present into the config file then broker will read it as false. also, sometimes user assumes that broker will pick default value if not present into config-file and in this case if broker falls to `tlsAllowInsecureConnection` then it will create unexpected behavior. Also, I don't see it a breaking change for future release so, we might just stay with this change?

Reply:
@rdhabalia if people already specify tlsAllowInsecureConnection to true, and when it upgrades to 2.3.0, he doesn't specify brokerClientTlsAllowInsecureConnection to true, then it is a breaking behavior, no? 

Either we should change concurrent configuration loading logic to have the ability to check if a setting is specified or not, or we should document this for people to know how to configure the behavior when upgrading to 2.3.0.